components/timezone/src/ids.rs

```diff
@@ -0,0 +1,677 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use alloc::borrow::Cow;
+use alloc::string::String;
+use alloc::vec::Vec;
+use icu_provider::prelude::*;
+use zerotrie::cursor::ZeroAsciiIgnoreCaseTrieCursor;
+
+use crate::{
+    provider::names::{
+        Bcp47ToIanaMapV1, Bcp47ToIanaMapV1Marker, IanaToBcp47MapV2, IanaToBcp47MapV2Marker,
+    },
+    TimeZoneBcp47Id, TimeZoneError,
+};
+
+/// A mapper between IANA time zone identifiers and BCP-47 time zone identifiers.
+///
+/// This mapper supports two-way mapping, but it is optimized for the case of IANA to BCP-47.
+/// It also supports normalizing and canonicalizing the IANA strings.
+///
+/// # Normalization vs Canonicalization
+///
+/// Multiple IANA time zone identifiers can refer to the same BCP-47 time zone. For example, the
+/// following three IANA identifiers all map to `"usind"`:
+///
+/// - "America/Fort_Wayne"
+/// - "America/Indiana/Indianapolis"
+/// - "America/Indianapolis"
+/// - "US/East-Indiana"
+///
+/// There is only one canonical name, which is "America/Indiana/Indianapolis". The
+/// *canonicalization* operation returns the canonical name. You should canonicalize if you
+/// need to compare time zones for equality or display the name to the user. Note that the
+/// canonical name can change over time.
```

Comment:
It might help to provide an example. 

```suggestion
/// canonical name can change over time, for example the identifier Europe/Kiev
/// was renamed to the newly-added identifier Europe/Kyiv in 2022.
```